usagov/usagov-2021
# Usagov 2117 phpcs local

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-2117

## Description

Adds PHPCodeSniffer and the Parallel PHP linter as dev tools for local environments.
Adds composer scripts to run the tools in the composer container.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

- run `composer install` to get the dependencies.
- run `./bin/composer phpcs-errors`. You should see a report from phpcs showing no errors
- run `./bin/composer php-lint`. You should see a report from phpcs showing no errors
- run `./bin/composer phpcs-errors-strict`. You should see a report from phpcs showing warnings across a lot of files

There are two commands that tell phpcs to only check files changed in the current branch, `phpcs-changes` and `phpcs-changes-strict`. If the others run, this one should run too. To confirm they work, you'd need to commit a change to some file in the project and run them. If you fix an existing code style issue, you shouldn't see it in the output. If you intentionally create one, it should be added to the report. You could make a commit and just not push any changes back up to here to be safe.
- Test instruction 3

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
